pkg/fake_pub_server/lib/fake_pub_server.dart

```diff
@@ -97,7 +102,10 @@ class FakePubServer {
         _logger.info('fake_pub_server running on port $port');
 
         await ProcessSignal.sigterm.watch().first;
+
+        _logger.info('fake_pub_server shutting down');
         await server.close();
+        nameTrackerUpdater.stop();
```

Comment:
Why not set this up in the service scope? so that it stops on exit from the service scope?

Reply:
Crazy idea:

`name_tracker.dart` should export a function called `setupNameTracker()` which will:
 * create a nametracker
 * start the name tracker
 * register the name tracker in the current service_scope
 * register stopping the name tracker in the current service_scope

Then we can call `setupNameTracker()` from `setupServices()` in `services.dart`. We might need a boolean to disable this on processes that doesn't need a name tracker running...